src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java

```diff
@@ -73,7 +73,7 @@
  * <li>
  * Property {@code groups} - Specify list of <b>type import</b> groups. Every group identified
  * either by a common prefix string, or by a regular expression enclosed in forward slashes
- * (e.g. {@code /regexp/}). All type imports, which does not match any group, falls into an
+ * (e.g. {@code /regexp/}). If an import matches two or more groups, the best match is selected. All type imports, which does not match any group, falls into an
```

Comment:
What does best actually mean here?

Reply:
It means whatever this algorithm does: https://github.com/checkstyle/checkstyle/blob/c1c73921b848af6ecc7ed7b2cc0c7f528d2f4d16/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java#L712-L726

I realize users might want to know what "best" means, and maybe someone will address that later; for now we're addressing the question of whether CheckStyle even considers this an acceptable use case.

Reply:
It looks like best is the closest to the start, and the longest match.

It looks like this is a regular expression find, and not a match. Non-regexp ensures we match the start of the string and an ending period.

Reply:
@JayBazuzi ,

`If an import matches two or more groups, the best match is selected (closest to the start, and the longest match)`

Does this sound good ?

Reply:
Looking at the resulting help page, the space is already overcrowded with text and adding that detail would make that problem worse.

If it was my project I would accept this change as-is as making things better, and then think about next steps like:
- no more change - this is enough detail to clue in users to the fact that multiple matches are OK
- extract all the common text to another place that goes in to detail, including examples

But it's not my project - you do what you want.

Reply:
Thanks a lot for sharing feedback.
Let me think, y will be merged until end of month for sure

Reply:
I think, it is better to update as I suggested, I can push update when I next time be around keyboard.